pkg/nas/controllerserver.go

```diff
@@ -235,56 +369,169 @@ func (cs *controllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
+		}
+		if !CheckNfsPathMounted(mountPoint, nfsServer, nfsPath) {
+			return nil, errors.New("Check Mount nfsserver fail " + nfsServer + " error with: ")
+		}
+		defer utils.Umount(mountPoint)
+
+		// pvName is same with volumeId
+		pvName := filepath.Base(pvPath)
+		deletePath := filepath.Join(mountPoint, pvName)
+		if _, err := os.Stat(deletePath); os.IsNotExist(err) {
+			log.Infof("Delete: Volume %s, Path %s does not exist, deletion skipped", req.VolumeId, deletePath)
+			return &csi.DeleteVolumeResponse{}, nil
+		}
+
+		// Determine if the "archiveOnDelete" parameter exists.
+		// If it exists and has a false value, delete the directory.
+		// Otherwise, archive it.
+		archiveOnDelete, exists := storageclass.Parameters["archiveOnDelete"]
+		if exists {
+			archiveBool, err := strconv.ParseBool(archiveOnDelete)
+			if err != nil {
+				return nil, errors.New("Check Mount nfsserver fail " + nfsServer + " error with: " + err.Error())
+			}
+			if !archiveBool {
+				if err := os.RemoveAll(deletePath); err != nil {
+					return nil, errors.New("Check Mount nfsserver fail " + nfsServer + " error with: " + err.Error())
+				}
+				log.Infof("Delete Successful: Volume %s, Removed path %s", req.VolumeId, deletePath)
+				return &csi.DeleteVolumeResponse{}, nil
+			}
+		}
+
+		archivePath := filepath.Join(mountPoint, "archived-"+pvName+time.Now().Format(".2006-01-02-15:04:05"))
+		if err := os.Rename(deletePath, archivePath); err != nil {
+			log.Errorf("Delete Failed: Volume %s, archiving path %s to %s with error: %s", req.VolumeId, deletePath, archivePath, err.Error())
+			return nil, errors.New("Check Mount nfsserver fail " + nfsServer + " error with: ")
+		}
+
+		log.Infof("Delete Successful: Volume %s, Archiving path %s to %s", req.VolumeId, deletePath, archivePath)
 	}
-	defer utils.Umount(mountPoint)
 
 	// remove the pvc process mapping if exist
 	if ok, _ := pvcProcessSuccess[req.VolumeId]; ok {
 		delete(pvcProcessSuccess, req.VolumeId)
 	}
 
-	// pvName is same with volumeId
-	pvName := filepath.Base(pvPath)
-	deletePath := filepath.Join(mountPoint, pvName)
-	if _, err := os.Stat(deletePath); os.IsNotExist(err) {
-		log.Infof("Delete: Volume %s, Path %s does not exist, deletion skipped", req.VolumeId, deletePath)
-		return &csi.DeleteVolumeResponse{}, nil
+	return &csi.DeleteVolumeResponse{}, nil
+}
+
+func (cs *controllerServer) getNasVolumeOptions(req *csi.CreateVolumeRequest) (*nasVolumeArgs, error) {
+	var ok bool
+	nasVolArgs := &nasVolumeArgs{}
+	volOptions := req.GetParameters()
+
+	if nasVolArgs.VolumeAs, ok = volOptions[VOLUMEAS]; !ok {
+		//return nil, fmt.Errorf("Required parameter [parameter.volumeAs] must be set in StorageClass, supported value is [filesystem] or [subpath]")
+		nasVolArgs.VolumeAs = "filesystem"
```

Comment:
it is better to default as subPath for compatible with early version.
And more customers can be statisfied by subPath mode.